pkg/services/mysqldb/bind_common.go

```diff
@@ -0,0 +1,95 @@
+package mysqldb
+
+import (
+	"fmt"
+
+	"github.com/Azure/open-service-broker-azure/pkg/generate"
+)
+
+func createBinding(
+	enforceSSL bool,
+	dnsSuffix string,
+	serverName string,
+	adminPassword string,
+	fqdn string,
+	databaseName string,
+) (*mysqlBindingDetails, error) {
+
+	userName := generate.NewIdentifier()
+	password := generate.NewPassword()
+
+	db, err := createDBConnection(
+		enforceSSL,
+		dnsSuffix,
+		serverName,
+		adminPassword,
+		fqdn,
+		databaseName,
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close() // nolint: errcheck
+
+	// Open doesn't open a connection. Validate DSN data:
+	if err = db.Ping(); err != nil {
+		return nil, err
+	}
+
+	if _, err = db.Exec(
+		fmt.Sprintf("CREATE USER '%s'@'%%' IDENTIFIED BY '%s'", userName, password),
+	); err != nil {
+		return nil, fmt.Errorf(
+			`error creating user "%s": %s`,
+			userName,
+			err,
+		)
+	}
+
+	if _, err = db.Exec(
+		fmt.Sprintf("GRANT SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, "+
+			"INDEX, ALTER, CREATE TEMPORARY TABLES, LOCK TABLES, "+
+			"CREATE VIEW, SHOW VIEW, CREATE ROUTINE, ALTER ROUTINE, "+
+			"EXECUTE, REFERENCES, EVENT, "+
+			"TRIGGER ON %s.* TO '%s'@'%%'",
+			databaseName, userName)); err != nil {
+		return nil, fmt.Errorf(
+			`error granting permission to "%s": %s`,
+			userName,
+			err,
+		)
+	}
+
+	return &mysqlBindingDetails{
+		LoginName: userName,
+		Password:  password,
+	}, nil
+
+}
+
+func createCredential(
+	fqdn string,
+	sslRequired bool,
+	serverName string,
+	databaseName string,
+	bindingDetails *mysqlBindingDetails,
+) *Credentials {
+	username := fmt.Sprintf("%s@%s", bindingDetails.LoginName, serverName)
+	connectionTemplate := "mysql://%s:%s@%s:3306/%s?allowNativePasswords=true"
```

Comment:
actually not sure if i should include allowNativePasswords here? I don't think that's a mysql option outside of the go lib?

Reply:
Not sure I know what that setting does...

Reply:
w/ the golang lib at least, if you *don't* include thatt mysql complains. I'll try it with java before i merge this and see what's up.

Reply:
Ok. lmk. Apart from this open question, this LGTM.